### PR TITLE
GH-167: Implement TypeScript Schedule Table

### LINKED ISSUE #167: Create TypeScript Schedule Schema & Table
Story Points: [\<Number of Development Days to complete task\>]

# How?
Add schedule table to TypeScript schema. It should have the following fields:
Unique ID for each Reservation: ID!
Who's reserving it (email): String!
What they're trying to reserve (ID of each item): String!
Reservation start date time: AWSDateTime!
Reservation end date time: AWSDateTime!
Optional reservation message: String
Make associated ScheduleTable.ts Class (See ItemsTable.ts or HistoryTable.ts for example)

### PR DESCRIPTION
**Pull request by johnnyhchen**
resolves #167

### CODE REVIEW

File: amplify/ts-code/src/db/Schemas.ts [hunk 1]
```diff
@@ -62,6 +62,24 @@ export interface HistorySchema {
...
+ * @param id Random Time Related Unique Key, where the first part of the key is the creation time.
+ * @param reserver Email of person reserving Item.
+ * @param itemId ID of Item.
+ * @param startTime Time reservation starts.
+ * @param endTime Time reservation end.
+ * @param notes Optional notes about this action.
+ */
+export interface ScheduleSchema {
+    id: string,
+    reserver: string,
```
Comment (yauj) [inline amplify/ts-code/src/db/Schemas.ts:76]: Can we make it consistent and make it `borrower`?

File: amplify/ts-code/src/db/ScheduleTable.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
+import { SCHEDULE_TABLE, ScheduleSchema, ITEMS_TABLE, ItemsSchema } from "./Schemas";
```
Comment (yauj) [inline amplify/ts-code/src/db/ScheduleTable.ts:1]: Hard for me to review this right now without tests.

**Review by yauj — CHANGES_REQUESTED** [review]

File: amplify/ts-code/src/db/Schemas.ts (hunk 1 above)
Comment (johnnyhchen) [reply amplify/ts-code/src/db/Schemas.ts]: Done!

File: amplify/ts-code/src/db/ScheduleTable.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+                        notes: notes
+                    }
+
+                    const indexParams: DocumentClient.PutItemInput = {
+                        TableName: SCHEDULE_TABLE,
+                        Item: reservation
+                    }
+
+                    return this.client.update(itemsParams)
+                        .then(() => this.client.put(indexParams))
```
Comment (yauj) [inline amplify/ts-code/src/db/ScheduleTable.ts:57]: I think you want to do a `put` before the `update`. If you're unable to put the object into the schedule table, then you don't want to append to the item table prematurely. Meanwhile, a orphaned `schedule` object, where there's no reference from item -> schedule is not the worst thing.

**Review by yauj — CHANGES_REQUESTED** [review]

File: amplify/ts-code/src/db/ScheduleTable.ts (hunk 3 above)
Comment (johnnyhchen) [reply amplify/ts-code/src/db/ScheduleTable.ts]: Gotcha!

**Review by yauj — APPROVED** [review]

**Pull request merged**